sympy/core/relational.py

```diff
@@ -302,17 +303,36 @@ def __new__(cls, lhs, rhs=0, **options):
             # If expressions have the same structure, they must be equal.
             if lhs == rhs:
                 return S.true
-
-            # If appropriate, check if the difference evaluates.  Detect
-            # incompatibility such as lhs real and rhs not real.
+            elif all(isinstance(i, BooleanAtom) for i in (rhs, lhs)):
+                return S.false
+
+            # check finiteness
+            fin = L, R = [i.is_finite for i in (lhs, rhs)]
+            if None not in fin:
+                if L != R:
+                    return S.false
+                if L is False:
+                    return S.true
+
+            # see if the difference evaluates
             if lhs.is_complex and rhs.is_complex:
                 r = (lhs - rhs).is_zero
                 if r is not None:
                     return _sympify(r)
-
-            # If expression have both Boolean terms
-            if all(isinstance(i, BooleanAtom) for i in (rhs, lhs)):
-                return S.false  # equal args already evaluated
+            elif all(not (isinstance(i,BooleanAtom) or isinstance(i, Tuple)) for i in (lhs,rhs)):
+                if (lhs - rhs).is_real is None:
```

Comment:
we need to check this for infinite numbers so i add this condition. this is not allowed for BooleanAtom and Tuple because S.true(lhs) - S.false(rhs) is not defined.